timesketch/ui/graphs/cytoscape-settings.component.spec.ts

```diff
@@ -0,0 +1,137 @@
+  beforeEach(async () => {
+    TestBed.configureTestingModule({
+      imports: [BrowserModule, FormsModule],
+      declarations: [CytoscapeComponent, CytoscapeSettingsComponent, SettingsTestComponent],
+    })
+    fixture = TestBed.createComponent(SettingsTestComponent)
+    // prevent 'Error: Timeout - Async callback was not invoked within timeout':
+    fixture.componentInstance.cytoscapeComponent['container'] = undefined
+    fixture.detectChanges()
+    await fixture.whenStable()
+  })
+
+  describe('checkboxes', () => {
+    it('should work correctly', () => {
+      for (const checkboxDebugElement of fixture.debugElement.queryAll(By.css('input[type="checkbox"]'))) {
+        const checkbox: HTMLInputElement = checkboxDebugElement.nativeElement
+        const label: HTMLElement = checkbox.parentElement
+        const labelText = label.textContent.trim()
+
+        expect(labelText in remainingOptions).toBeTruthy()
+        delete remainingOptions[labelText]
+
+        if (labelText === 'headless' || labelText === 'styleEnabled') {
+          // these cause PhantomJS to hang
+          continue
+        }
+        label.click()
+        fixture.detectChanges()
+
+        const cy = fixture.componentInstance.cytoscapeComponent.cy
+        expect(Boolean(getOptionFromCytoscape(cy, labelText))).toEqual(!testCytoscapeOptions[labelText])
+      }
+    })
+  })
+
+  describe('number inputs', () => {
+    it('should work correctly', () => {
+      for (const inputDebugElement of fixture.debugElement.queryAll(By.css('input[type="number"]'))) {
+        const input: HTMLInputElement = inputDebugElement.nativeElement
+        const label: HTMLElement = input.parentElement
+        const labelText = label.textContent.trim()
+        const value = testCytoscapeOptions[labelText]
+
+        expect(labelText in remainingOptions).toBeTruthy()
+        delete remainingOptions[labelText]
+
+        input.value = String(value + 3)
+        input.dispatchEvent(new Event('input'))
+        fixture.detectChanges()
+
+        const cy = fixture.componentInstance.cytoscapeComponent.cy
+        expect(getOptionFromCytoscape(cy, labelText)).toEqual(value + 3)
+      }
+    })
+  })
+
+  describe('radio buttons', () => {
+    it('should work correctly', () => {
+      for (const inputDebugElement of fixture.debugElement.queryAll(By.css('input[type="radio"]'))) {
+        const input: HTMLInputElement = inputDebugElement.nativeElement
+        const label: HTMLElement = input.parentElement
+        const name = input.name
+        const value = input.value
+
+        delete remainingOptions[name]
+
+        label.click()
+        fixture.detectChanges()
+
+        const cy = fixture.componentInstance.cytoscapeComponent.cy
+        expect(getOptionFromCytoscape(cy, name)).toEqual(value)
+      }
+    })
+  })
+
+  it('should have tests covering all cytoscape options', () => {
+    expect(Object.keys(remainingOptions).length).toEqual(0)
```

Comment:
This makes your tests depend on each other and easier to break. If someone adds another test or moves them around, they will start to fail. I proposed a different solution in a comment further up.

Reply:
Yes, this makes them order-dependent, but they are guaranted to execute in order they are defined. Solution from your comment is less dry, so it is not clear which one is better.